File: app/src/main/java/me/bilalhaider/moviedatabase/ui/movie/MovieActivity.java
package me.bilalhaider.moviedatabase.ui.movie;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import butterknife.ButterKnife;
import me.bilalhaider.moviedatabase.R;
import me.bilalhaider.moviedatabase.data.model.MovieResponse;
import me.bilalhaider.moviedatabase.util.CommonFunctions;

public class MovieActivity extends AppCompatActivity implements MovieContract.View {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public Boolean isNetworkAvailable() {
        return CommonFunctions.isNetworkAvailable(this);
    }

    @Override
    public void showProgress() {

    }

    @Override
    public void hideProgress() {

    }

    @Override
    public void handleOnResponseSuccess(MovieResponse movieResponse) {

    }

    @Override
    public void handleOnErrorResponse(String errorMessage) {

    }
}

File: app-android/src/main/java/me/bilalhaider/moviedatabase/movie/MovieActivity.kt
package me.bilalhaider.moviedatabase.movie

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.material.MaterialTheme
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.viewmodel.compose.viewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.launch

/**
 * Created by Bilal Haider on 17/03/2022
 */
@AndroidEntryPoint
class MovieActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_MOVIE_ID = "movie_id"

        fun start(
            context: Context,
            movieID: String
        ) {
            context.startActivity(
                with(Intent(context, MovieActivity::class.java)) {
                    putExtra(EXTRA_MOVIE_ID, movieID)
                }
            )
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        intent.getStringExtra(EXTRA_MOVIE_ID)?.let {
            createUI(it)
        } ?: finish()
    }

    private fun createUI(movieId: String) {
        lifecycleScope.launch {
            setContent {

                viewModel<MovieViewModel>().apply {
                    fetchMovie(movieId)
                }

                MaterialTheme {
                    Movie()
                }

            }
        }

    }

}